mne/epochs.py

```diff
@@ -111,12 +112,7 @@ def _save_split(epochs, fname, part_idx, n_parts, fmt):
     # Metadata
     if epochs.metadata is not None:
         start_block(fid, FIFF.FIFFB_MNE_METADATA)
-        metadata = epochs.metadata
-        if not isinstance(metadata, list):
-            metadata = metadata.to_json(orient='records')
-        else:  # Pandas DataFrame
-            metadata = json.dumps(metadata)
-        assert isinstance(metadata, str)
+        metadata = epochs._prepare_write_metadata()
```

Comment:
I would make _prepare_write_metadata a private function like you did for _check_event_id to avoid adding a new method.

`metadata = _prepare_write_metadata(self.metadata)`